packages/eslint-plugin/rules/prefer-typescript.js

```diff
@@ -0,0 +1,26 @@
+const path = require('node:path');
+
+const { getDocURL } = require('../lib/utils');
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Prefer using TypeScript within a codebase.',
+      url: getDocURL(__filename),
+    },
+  },
+  create: context => {
+    return {
+      Program(node) {
```

Comment:
Though I can see if a file actually contains TS by checking if it uses `import` or some sort of type but because a file could potentially have neither of those, and still be perfectly fine, I'm leaving it to just assert the filename.

Though maybe in those cases the warning could just be ignored? I'm not sure.